sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/ObjectReplicationPolicy.java

```diff
@@ -3,24 +3,24 @@
 
 package com.azure.storage.blob.models;
 
-import java.util.HashMap;
-import java.util.Map;
+import java.util.ArrayList;
+import java.util.List;
 
 /**
  * A type that contains information about an object replication policy on a source blob.
  */
 public class ObjectReplicationPolicy {
 
     private final String policyId;
-    private final Map<String, String> ruleStatuses;
+    private final List<ObjectReplicationRule> objectReplicationRules;
 
     ObjectReplicationPolicy(String policyId) {
         this.policyId = policyId;
-        this.ruleStatuses = new HashMap<>();
+        this.objectReplicationRules = new ArrayList<>();
     }
```

Comment:
We should accept list of rules as constructor parameter as well, so that this is fully functional after construction (avoid multi-step init).
If we're aiming to make this class immutable we should clone the list (https://stackoverflow.com/questions/6137224/immutable-object-with-arraylist-member-variable-why-can-this-variable-be-chang )

Reply:
Should I make all the constructors public?

Reply:
Yes, customers should be able to create these types for test purposes. See `BlobContainerAccessPolicies` as an example (though that one isn't perfectly immutable...).